fili-core/src/main/java/com/yahoo/bard/webservice/util/IntervalUtils.java

```diff
@@ -209,4 +211,20 @@ public static long getTotalDuration(Collection<Interval> intervals) {
                 .map(Interval::getStart)
                 .reduce(Utils::getMinValue);
     }
+
+    /**
+     * Returns the coarsest ZonedTimeGrain among a set of ZonedTimeGrains.
+     * <p>
+     * If the set of ZonedTimeGrains is empty, return null.
+     *
+     * @param zonedTimeGrains  A set of ZonedTimeGrains among which the coarsest ZonedTimeGrain is to be returned.
+     *
+     * @return the coarsest ZonedTimeGrain among a set of ZonedTimeGrains
+     */
+    public static ZonedTimeGrain getCoarsestTimeGrain(Collection<ZonedTimeGrain> zonedTimeGrains) {
+        return zonedTimeGrains.stream()
+                .sorted(Comparator.comparing(ZonedTimeGrain::getEstimatedDuration))
```

Comment:
@cdeszaq Because `GranularityComparator` compares by taking `PhysicalTable` not `ZonedTimeGrain`. I can write a comparator for `ZonedTimeGrain` if that's you would like to do.